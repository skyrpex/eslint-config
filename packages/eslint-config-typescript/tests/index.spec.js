const { createEngine, cleanReport } = require("../../../tests/utils");

const engine = createEngine({
    configFile: require.resolve("../index.js"),
});

test("should report missing trailing slash", () => {
    const report = engine.executeOnFiles(["typescript.fixture.ts"]);
    expect(cleanReport(report)).toMatchSnapshot();
});
