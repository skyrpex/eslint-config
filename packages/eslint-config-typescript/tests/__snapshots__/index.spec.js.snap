// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should report missing trailing slash 1`] = `
Object {
  "errorCount": 5,
  "fixableErrorCount": 4,
  "fixableWarningCount": 0,
  "results": Array [
    Object {
      "errorCount": 5,
      "filePath": "tests/fixtures/typescript.fixture.ts",
      "fixableErrorCount": 4,
      "fixableWarningCount": 0,
      "messages": Array [
        Object {
          "column": 7,
          "endColumn": 19,
          "endLine": 1,
          "line": 1,
          "message": "'test' is assigned a value but never used.",
          "nodeType": "Identifier",
          "ruleId": "no-unused-vars",
          "severity": 2,
          "source": "const test: number = \\"test\\"",
        },
        Object {
          "column": 28,
          "endColumn": 28,
          "endLine": 1,
          "fix": Object {
            "range": Array [
              27,
              27,
            ],
            "text": ";",
          },
          "line": 1,
          "message": "Insert \`;\`",
          "nodeType": null,
          "ruleId": "prettier/prettier",
          "severity": 2,
          "source": "const test: number = \\"test\\"",
        },
        Object {
          "column": 17,
          "endColumn": 17,
          "endLine": 4,
          "fix": Object {
            "range": Array [
              79,
              79,
            ],
            "text": ";",
          },
          "line": 4,
          "message": "Insert \`;\`",
          "nodeType": null,
          "ruleId": "prettier/prettier",
          "severity": 2,
          "source": "    name: string",
        },
        Object {
          "column": 21,
          "endColumn": 21,
          "endLine": 5,
          "fix": Object {
            "range": Array [
              100,
              100,
            ],
            "text": ";",
          },
          "line": 5,
          "message": "Insert \`;\`",
          "nodeType": null,
          "ruleId": "prettier/prettier",
          "severity": 2,
          "source": "    age: number = 18",
        },
        Object {
          "column": 31,
          "endColumn": 31,
          "endLine": 8,
          "fix": Object {
            "range": Array [
              152,
              152,
            ],
            "text": ";",
          },
          "line": 8,
          "message": "Insert \`;\`",
          "nodeType": null,
          "ruleId": "prettier/prettier",
          "severity": 2,
          "source": "        console.log(this.name)",
        },
      ],
      "source": "const test: number = \\"test\\"

export default class DummyClass {
    name: string
    age: number = 18

    constructor() {
        console.log(this.name)
    }
}
",
      "warningCount": 0,
    },
  ],
  "warningCount": 0,
}
`;
